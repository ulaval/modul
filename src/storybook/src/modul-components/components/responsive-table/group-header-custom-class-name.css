.test-group-header-custom-class-name.test-group-header-custom-class-name {
    border-left: 4px solid #000;
    background: #ccc;
}

.test-group-header-custom-class-name.test-group-header-custom-class-name th {
    padding: 8px 16px;
}

.test-icon-accordion-custom-class-name.test-icon-accordion-custom-class-name {
    color: blue;
}

.test-cell-custom-class-name.test-cell-custom-class-name {
    background: #000;
    color: #fff;
}
