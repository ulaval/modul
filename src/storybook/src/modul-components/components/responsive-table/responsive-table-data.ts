import { MTableAccordion, MTableAccordionIconPosition, MTableColspan, MTableColumn, MTableEmptyArea, MTableHeader, MTableHeadRows, MTableRow, MTableRowsGroup, MTableSortDirection, MTableTextAlign } from '@ulaval/modul-components/dist/components/responsive-table/responsive-table-commons';
import { ModulIconName } from '@ulaval/modul-components/dist/utils/modul-icons/modul-icons';

export const getTableColumn: (
    name: string,
    value: string,
    sortable?: boolean,
    sortDirection?: MTableSortDirection,
    defaultSortDirection?: MTableSortDirection,
    textAlign?: MTableTextAlign
) => MTableColumn = (
    name: string,
    value: string,
    sortable: boolean = true,
    sortDirection: MTableSortDirection | undefined = undefined,
    defaultSortDirection: MTableSortDirection = MTableSortDirection.Asc,
    textAlign: MTableTextAlign = MTableTextAlign.Left

) => {
    return {
        name,
        value,
        sortable,
        enableUnsort: true,
        textAlign,
        sortDirection,
        defaultSortDirection
    };
};

export const getScopeSlotTemplate = (
    textHtml: string
) => {
    return `<div
            style="background: blue; color: #fff; padding: 8px; border-radius: 4px;">
            ${ textHtml}
        </div>`;
};

export const MAIN_ROW: string = 'mainRow';

const getDefaultTableColumns = (isSortable: boolean = true) => {
    return [
        getTableColumn('name', 'Name', isSortable, MTableSortDirection.Asc),
        getTableColumn('id', 'Id', isSortable, MTableSortDirection.None, MTableSortDirection.None, MTableTextAlign.Center),
        getTableColumn('email', 'Email', isSortable),
        getTableColumn('dateOfBirth', 'Date of birth', isSortable, MTableSortDirection.None, MTableSortDirection.Dsc),
        getTableColumn('phoneNumber', 'Phone number', isSortable)
    ];
};

export const DEFAULT_TABLE_COLUMNS: MTableColumn[] = getDefaultTableColumns(true);
export const DEFAULT_TABLE_COLUMNS_DISABLED_SORTABLE: MTableColumn[] = getDefaultTableColumns(false);

export const DEFAULT_EMPTY_AREA: MTableEmptyArea = {
    title: 'No data to display',
    svgName: ModulIconName.FolderOpen,
    buttonText: 'Add content'
};

export const DEFAULT_TABLE_ROW_1: MTableRow = {
    cells: {
        name: {
            value: 'Name 1',
            className: 'add-custom-class-cell',
            isHeader: true
        },
        id: {
            value: '111 222 333'
        },
        email: {
            value: 'fake1@email.com'
        },
        dateOfBirth: {
            value: '1971-12-05'
        },
        phoneNumber: {
            value: '819 555-0111'
        }
    }
};

export const DEFAULT_TABLE_ROW_2: MTableRow = {
    cells: {
        name: {
            value: 'Name 2',
            data: {
                nom: 'First Name',
                prenom: 'Family Name'
            },
            isHeader: true
        },
        id: {
            value: '111 322 344'
        },
        email: {
            value: 'fake2@email.com'
        },
        dateOfBirth: {
            value: '1980-06-11'
        },
        phoneNumber: {
            value: '418 555-0163'
        }
    },
    className: 'add-custom-class-row'
};

export const DEFAULT_TABLE_ROW_3: MTableRow = {
    cells: {
        name: {
            value: 'Name 3',
            isHeader: true
        },
        id: {
            value: '112 528 967'
        },
        email: {
            value: 'fake3@email.com'
        },
        dateOfBirth: {
            value: '1970-04-08'
        },
        phoneNumber: {
            value: '438 555-0113'
        }
    }
};

export const DEFAULT_TABLE_GROUP_HEADER_1: MTableHeader = {
    title: 'English Class',
    cells: {
        name: {
            value: 'ANL-3010'
        },
        id: {
            value: 'Advanced English I',
            colspan: 4,
            textAlign: MTableTextAlign.Right
        }
    }
};

export const DEFAULT_TABLE_GROUP_HEADER_2: MTableHeader = {
    title: 'Title',
    cells: {
        name: {
            value: 'Professor\'s name',
            colspan: MTableColspan.AllColumns
        }
    }
};

export const DEFAULT_TABLE_GROUP_HEADER_3: MTableHeader = {
    cells: {
        name: {
            value: 'ANI-1101'
        },
        id: {
            value: 'Motion design',
            colspan: 4,
            textAlign: MTableTextAlign.Left
        }
    }
};

export const DEFAULT_TABLE_GROUP_HEADER_4: MTableHeader = {
    cells: {
        name: {
            value: 'DES-1103'
        },
        id: {
            value: 'User experience design I',
            colspan: 4,
            textAlign: MTableTextAlign.Left
        }
    }
};

export const DEFAULT_TABLE_GROUP_HEADER_5: MTableHeader = {
    cells: {
        name: {
            value: 'DES-1104'
        },
        id: {
            value: 'User experience design II',
            colspan: 4,
            textAlign: MTableTextAlign.Left
        }
    }
};

export const DEFAULT_TABLE_HEAD_ROWS: MTableHeadRows = {
    [MAIN_ROW]: {
        columns: DEFAULT_TABLE_COLUMNS,
        mainColumns: true
    }
};

export const DEFAULT_TABLE_GROUP_ACCORDION_1: MTableAccordion = {
    open: true,
    disabled: false,
    iconPosition: MTableAccordionIconPosition.Left,
    iconClassName: 'add-class-to-icon'
};

export const DEFAULT_TABLE_GROUP_ACCORDION_2: MTableAccordion = {
    open: false,
    disabled: true
};

export const DEFAULT_TABLE_GROUP_ACCORDION_3: MTableAccordion = {
    open: true,
    displayIcon: false
};

export const DEFAULT_TABLE_ROWS_GROUP_1: MTableRowsGroup = {
    name: 'group1',
    accordion: DEFAULT_TABLE_GROUP_ACCORDION_1,
    header: DEFAULT_TABLE_GROUP_HEADER_1,
    rows: [
        DEFAULT_TABLE_ROW_1,
        DEFAULT_TABLE_ROW_2,
        DEFAULT_TABLE_ROW_3
    ]
};

export const DEFAULT_TABLE_ROWS_GROUP_2: MTableRowsGroup = {
    name: 'group2',
    header: DEFAULT_TABLE_GROUP_HEADER_2,
    accordion: DEFAULT_TABLE_GROUP_ACCORDION_3,
    rows: [
        DEFAULT_TABLE_ROW_1
    ]
};

export const DEFAULT_TABLE_ROWS_GROUP_3: MTableRowsGroup = {
    name: 'group3',
    header: DEFAULT_TABLE_GROUP_HEADER_3,
    accordion: DEFAULT_TABLE_GROUP_ACCORDION_1,
    rows: [
        DEFAULT_TABLE_ROW_1,
        DEFAULT_TABLE_ROW_2,
        DEFAULT_TABLE_ROW_3
    ]
};

export const DEFAULT_TABLE_ROWS_GROUP_4: MTableRowsGroup = {
    name: 'group4',
    header: DEFAULT_TABLE_GROUP_HEADER_4,
    accordion: DEFAULT_TABLE_GROUP_ACCORDION_2,
    rows: [
        DEFAULT_TABLE_ROW_1,
        DEFAULT_TABLE_ROW_2,
        DEFAULT_TABLE_ROW_3
    ]
};

export const DEFAULT_TABLE_ROWS_GROUP_5: MTableRowsGroup = {
    name: 'group5',
    header: DEFAULT_TABLE_GROUP_HEADER_5,
    rows: [
        DEFAULT_TABLE_ROW_1,
        DEFAULT_TABLE_ROW_2
    ]
};

export const DEFAULT_TABLE_ROWS_GROUP_6: MTableRowsGroup = {
    name: 'group6',
    rows: [
        DEFAULT_TABLE_ROW_1,
        DEFAULT_TABLE_ROW_2,
        DEFAULT_TABLE_ROW_2
    ]
};

export const DEFAULT_TABLE_ROWS_GROUP_EMPTY_1: MTableRowsGroup = {
    name: 'groupEmpty1',
    header: {
        title: 'Group empty'
    },
    rows: [],
    accordion: {
        open: true
    },
    emptyArea: {
        title: 'No data',
        svgName: ModulIconName.Grid,
        buttonText: 'Add row'
    }
};

export const DEFAULT_TABLE_ROWS: MTableRow[] = [
    DEFAULT_TABLE_ROW_1,
    DEFAULT_TABLE_ROW_2,
    DEFAULT_TABLE_ROW_3
];

export const DEFAULT_TABLE_ROW_GROUPS: MTableRowsGroup[] = [
    DEFAULT_TABLE_ROWS_GROUP_1,
    DEFAULT_TABLE_ROWS_GROUP_2
];

export const DEFAULT_TABLE_ROW_GROUPS_2: MTableRowsGroup[] = [
    DEFAULT_TABLE_ROWS_GROUP_2,
    DEFAULT_TABLE_ROWS_GROUP_3,
    DEFAULT_TABLE_ROWS_GROUP_4,
    DEFAULT_TABLE_ROWS_GROUP_5,
    DEFAULT_TABLE_ROWS_GROUP_6
];

export const COMPLEX_TABLE_FOLDER_COLUMNS: MTableColumn[] = [
    {
        name: 'forlderSpacingColumn',
        value: '',
        order: 1,
        rowspan: 2,
        style: { background: '#fff' }
    },
    {
        name: 'forlderBColumn',
        value: 'Folder B',
        order: 3,
        colspan: 4,
        textAlign: MTableTextAlign.Center
    },
    {
        name: 'forlderAColumn',
        value: 'Folder A',
        order: 2,
        colspan: 4,
        textAlign: MTableTextAlign.Center
    }
];

export const COMPLEX_TABLE_AFTER_BEFORE_COLUMNS: MTableColumn[] = [
    {
        name: 'beforeForlderAColumn',
        value: 'Before',
        colspan: 2,
        textAlign: MTableTextAlign.Center
    },
    {
        name: 'afterForlderAColumn',
        value: 'After',
        colspan: 2,
        textAlign: MTableTextAlign.Center
    },
    {
        name: 'beforeForlderBColumn',
        value: 'Before',
        colspan: 2,
        textAlign: MTableTextAlign.Center
    },
    {
        name: 'afterForlderBColumn',
        value: 'After',
        colspan: 2,
        textAlign: MTableTextAlign.Center
    }
];

export const COMPLEX_TABLE_ID_COLUMN: string = 'userNaimeBeforeFolderA';
export const COMPLEX_TABLE_USER_NAME_BEFORE_FOLDER_A_COLUMN_NAME: string = 'userNameBeforeFolderA';
export const COMPLEX_TABLE_NAME_BEFORE_FOLDER_A_COLUMN_NAME: string = 'userBeforeFolderA';
export const COMPLEX_TABLE_USER_NAME_AFTER_FOLDER_A_COLUMN_NAME: string = 'userNameAfterFolderA';
export const COMPLEX_TABLE_NAME_AFTER_FOLDER_A_COLUMN_NAME: string = 'userAfterFolderA';
export const COMPLEX_TABLE_USER_NAME_BEFORE_FOLDER_B_COLUMN_NAME: string = 'userNameBeforeFolderB';
export const COMPLEX_TABLE_NAME_BEFORE_FOLDER_B_COLUMN_NAME: string = 'userBeforeFolderB';
export const COMPLEX_TABLE_USER_NAME_AFTER_FOLDER_B_COLUMN_NAME: string = 'userNameAfterFolderB';
export const COMPLEX_TABLE_NAME_AFTER_FOLDER_B_COLUMN_NAME: string = 'userAfterFolderB';


export const COMPLEX_TABLE_MAIN_COLUMNS: MTableColumn[] = [
    {
        name: COMPLEX_TABLE_ID_COLUMN,
        value: 'Id',
        width: '120px',
        textAlign: MTableTextAlign.Center
    },
    getTableColumn(COMPLEX_TABLE_USER_NAME_BEFORE_FOLDER_A_COLUMN_NAME, 'User Name', false),
    getTableColumn(COMPLEX_TABLE_NAME_BEFORE_FOLDER_A_COLUMN_NAME, 'Name', false),
    getTableColumn(COMPLEX_TABLE_USER_NAME_AFTER_FOLDER_A_COLUMN_NAME, 'User Name', false),
    getTableColumn(COMPLEX_TABLE_NAME_AFTER_FOLDER_A_COLUMN_NAME, 'Name', false),
    getTableColumn(COMPLEX_TABLE_USER_NAME_BEFORE_FOLDER_B_COLUMN_NAME, 'User Name', false),
    getTableColumn(COMPLEX_TABLE_NAME_BEFORE_FOLDER_B_COLUMN_NAME, 'Name', false),
    getTableColumn(COMPLEX_TABLE_USER_NAME_AFTER_FOLDER_B_COLUMN_NAME, 'User Name', false),
    getTableColumn(COMPLEX_TABLE_NAME_AFTER_FOLDER_B_COLUMN_NAME, 'Name', false)
];

export const getComplexTableRow = (
    id: string,
    userNameBeforeFolderA: string,
    nameBeforeFolderA: string,
    userNameAfterFolderA: string,
    nameAfterFolderA: string,
    userNameBeforeFolderB: string,
    nameBeforeFolderB: string,
    userNameAfterFolderB: string,
    nameAfterFolderB: string
) => {
    return {
        [COMPLEX_TABLE_ID_COLUMN]: {
            value: id,
            isHeader: true
        },
        [COMPLEX_TABLE_USER_NAME_BEFORE_FOLDER_A_COLUMN_NAME]: {
            value: userNameBeforeFolderA
        },
        [COMPLEX_TABLE_NAME_BEFORE_FOLDER_A_COLUMN_NAME]: {
            value: nameBeforeFolderA
        },
        [COMPLEX_TABLE_USER_NAME_AFTER_FOLDER_A_COLUMN_NAME]: {
            value: userNameAfterFolderA
        },
        [COMPLEX_TABLE_NAME_AFTER_FOLDER_A_COLUMN_NAME]: {
            value: nameAfterFolderA
        },
        [COMPLEX_TABLE_USER_NAME_BEFORE_FOLDER_B_COLUMN_NAME]: {
            value: userNameBeforeFolderB
        },
        [COMPLEX_TABLE_NAME_BEFORE_FOLDER_B_COLUMN_NAME]: {
            value: nameBeforeFolderB
        },
        [COMPLEX_TABLE_USER_NAME_AFTER_FOLDER_B_COLUMN_NAME]: {
            value: userNameAfterFolderB

        },
        [COMPLEX_TABLE_NAME_AFTER_FOLDER_B_COLUMN_NAME]: {
            value: nameAfterFolderB
        }
    };
};

export const COMPLEX_TABLE_ROWS: MTableRow[] = [
    {
        cells: getComplexTableRow('01', 'simon-test', 'Simons', 'simon.ca', 'Simon', 'test.simon', 'Simone', 'simon-com', 'Simon')
    },
    {
        cells: getComplexTableRow('02', '20-100', 'Vincen', '_vincent', 'Vincent', 'test.vincent', 'VINcent', 'vincent_g', 'Vincent')
    },
    {
        cells: getComplexTableRow('03', 'maxime-ber', 'maxime', 'maxim.ca', 'Maxim', 'max.max', 'MAXime', 'maxim.ber', 'Maxime')
    },
    {
        cells: getComplexTableRow('04', 'frankD', 'Frank', 'frank.ca', 'Francois', 'JF', 'Jean', 'JFrancois', 'Francois')
    },
    {
        cells: getComplexTableRow('05', 'Will.B', 'Will', 'simon.ca', 'William', 'b.will', 'willy', 'william.b', 'William')
    },
    {
        cells: getComplexTableRow('06', 'johny_02', 'john', 'john-02', 'John', '02.jo', 'John', 'jo.02', 'John')
    }
];

export const COMPLEX_TABLE_HEAD_ROWS: MTableHeadRows = {
    'folderRow': {
        columns: COMPLEX_TABLE_FOLDER_COLUMNS
    },
    'beforeAfterRow': {
        columns: COMPLEX_TABLE_AFTER_BEFORE_COLUMNS
    },
    [MAIN_ROW]: {
        columns: COMPLEX_TABLE_MAIN_COLUMNS,
        mainColumns: true
    }
};

export const COLUMNS: MTableColumn[] = [
    {
        name: 'name',
        value: 'Name',
        sortable: true,
        width: '300px',
        textAlign: MTableTextAlign.Left,
        sortDirection: MTableSortDirection.None,
        defaultSortDirection: MTableSortDirection.None
    },
    {
        name: 'id',
        value: 'Id',
        sortable: true,
        textAlign: MTableTextAlign.Left,
        className: 'add-class-id',
        sortDirection: MTableSortDirection.None,
        defaultSortDirection: MTableSortDirection.None
    },
    {
        name: 'nrc',
        value: 'NRC',
        sortable: true,
        enableUnsort: false,
        textAlign: MTableTextAlign.Center,
        sortDirection: MTableSortDirection.None,
        defaultSortDirection: MTableSortDirection.None
    },
    {
        name: 'program',
        value: 'Program',
        sortable: true,
        enableUnsort: false,
        textAlign: MTableTextAlign.Left,
        className: 'add-class-program',
        sortDirection: MTableSortDirection.None,
        defaultSortDirection: MTableSortDirection.None
    },
    {
        name: 'date',
        value: 'Date',
        sortable: true,
        enableUnsort: false,
        textAlign: MTableTextAlign.Right,
        sortDirection: MTableSortDirection.None,
        defaultSortDirection: MTableSortDirection.None
    }
];

export const HEAD_ROW: MTableHeadRows = {
    'row1': {
        order: 3,
        columns: COLUMNS
    },
    'row2': {
        order: 1,
        columns: [
            {
                name: 'full-name',
                value: 'name',
                order: 5
            },
            {
                name: 'other-info',
                value: 'Other infos',
                colspan: 4,
                order: 1
            }
        ]
    }
};

export const ROWS_GROUP: MTableRowsGroup = {
    name: 'ROWS_GROUP',
    accordion: {
        open: true,
        disabled: false,
        iconPosition: MTableAccordionIconPosition.Left
    },
    header: {
        title: 'Nom du cours 2',
        cells: {
            name: {
                value: 'name'
            },
            id: {
                value: 'id'
            },
            nrc: {
                value: 'nrc',
                colspan: 3
            }
        }
    },
    rows: [
        {
            cells: {
                name: {
                    value: 'Vincent',
                    className: 'ajouter-une-class-test'
                },
                id: {
                    value: '111 222 333'
                },
                nrc: {
                    value: '84 533'
                },
                program: {
                    value: 'Baccalauréat en design graphique'
                },
                date: {
                    value: '2020-06-05'
                }
            }
        },
        {
            cells: {
                name: {
                    value: 'Maxime'
                },
                id: {
                    value: '111 222 333'
                },
                nrc: {
                    value: '84 533'
                },
                program: {
                    value: 'Baccalauréat en design graphique'
                },
                date: {
                    value: '2020-06-05'
                }
            }
        }
    ],
    emptyArea: {
        title: 'No data'
    }
};

export const ROWS_GROUP_2: MTableRowsGroup = {
    name: 'ROWS_GROUP_2',
    accordion: {
        open: true,
        disabled: false,
        iconPosition: MTableAccordionIconPosition.Left
    },
    header: {
        title: 'Entete 2',
        cells: {
            name: {
                value: 'Entete 2 Martin'
            },
            id: {
                value: '111 222 333',
                colspan: 4
            }
        }
    },
    rows: [
        {
            cells: {
                name: {
                    value: 'Simard, Martin',
                    className: 'ajouter-une-class-test'
                },
                id: {
                    value: '111 222 333'
                },
                nrc: {
                    value: '84 533'
                },
                program: {
                    value: 'Baccalauréat en design graphique'
                },
                date: {
                    value: '2020-06-05'
                }
            }
        },
        {
            cells: {
                name: {
                    value: 'Alex',
                    className: 'ajouter-une-class-test'
                },
                id: {
                    value: '111 222 333'
                },
                nrc: {
                    value: '84 533'
                },
                program: {
                    value: 'Baccalauréat en design graphique'
                },
                date: {
                    value: '2020-06-05'
                }
            }
        },
        {
            cells: {
                name: {
                    value: 'Simon',
                    className: 'ajouter-une-class-test'
                },
                id: {
                    value: '111 222 333'
                },
                nrc: {
                    value: '84 533'
                },
                program: {
                    value: 'Baccalauréat en design graphique'
                },
                date: {
                    value: '2020-06-05'
                }
            }
        },
        {
            cells: {
                name: {
                    value: 'Vincent',
                    className: 'ajouter-une-class-test'
                },
                id: {
                    value: '111 222 333'
                },
                nrc: {
                    value: '84 533'
                },
                program: {
                    value: 'Baccalauréat en design graphique'
                },
                date: {
                    value: '2020-06-05'
                }
            }
        }
    ]
};

export const ROW_GROUPS: MTableRowsGroup[] = [ROWS_GROUP, ROWS_GROUP_2];

export const ROWS: MTableRow[] = [
    {
        cells: {
            name: {
                value: 'Titi',
                className: 'add-class'
            },
            id: {
                value: '111 222 333'
            },
            nrc: {
                value: '1234'
            },
            programme: {
                value: 'Baccalauréat en design graphique'
            },
            dateDerniereModification: {
                value: '2020-06-05'
            }
        }
    },
    {
        cells: {
            name: {
                value: 'Alex',
                className: 'ajouter-une-class-test'
            },
            id: {
                value: '111 222 333'
            },
            nrc: {
                value: '1234'
            },
            programme: {
                value: 'Baccalauréat en design graphique'
            },
            dateDerniereModification: {
                value: '2020-06-05'
            }
        }
    }
];
