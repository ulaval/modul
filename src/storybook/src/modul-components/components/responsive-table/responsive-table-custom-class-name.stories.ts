import { select } from '@storybook/addon-knobs';
import { RESPONSIVE_TABLE_NAME } from '@ulaval/modul-components/dist/components/component-names';
import { MTableGroupHeaderStyle, MTableRowsStyle } from '@ulaval/modul-components/dist/components/responsive-table/responsive-table-commons';
import { Enums } from '@ulaval/modul-components/dist/utils/enums/enums';
import { modulComponentsHierarchyRootSeparator } from '../../../utils';
import './group-header-custom-class-name.css';
import { DEFAULT_TABLE_COLUMNS_DISABLED_SORTABLE, DEFAULT_TABLE_ROW_1, DEFAULT_TABLE_ROW_2, DEFAULT_TABLE_ROW_3 } from './responsive-table-data';

export default {
    title: `${modulComponentsHierarchyRootSeparator}${RESPONSIVE_TABLE_NAME}/Custom class-name`,
    parameters: { fileName: __filename }
};

export const GroupHeader = () => ({
    data: () => ({
        columns: DEFAULT_TABLE_COLUMNS_DISABLED_SORTABLE,
        rowGroups: [
            {
                name: 'CustomStyle',
                header: {
                    title: 'Group #1',
                    className: 'test-group-header-custom-class-name'
                },
                accordion: {
                    open: true,
                    iconClassName: 'test-icon-accordion-custom-class-name'
                },
                rows: [
                    DEFAULT_TABLE_ROW_1,
                    DEFAULT_TABLE_ROW_2,
                    DEFAULT_TABLE_ROW_3
                ]
            }
        ]
    }),
    props: {
        groupHeaderStyle: {
            default: select(
                'Prop group-header-style',
                Enums.toValueArray(MTableGroupHeaderStyle),
                MTableGroupHeaderStyle.Any
            )
        },
        rowsStyle: {
            default: select(
                'Prop rows-style',
                Enums.toValueArray(MTableRowsStyle),
                MTableRowsStyle.Borders
            )
        }
    },
    template: `<${RESPONSIVE_TABLE_NAME}
        id="GroupHeader"
        :columns="columns"
        :row-groups="rowGroups"
        :group-header-style="groupHeaderStyle"
        :rows-style="rowsStyle"
    />`
});

export const Cells = () => ({
    data: () => ({
        columns: DEFAULT_TABLE_COLUMNS_DISABLED_SORTABLE,
        rows: [
            {
                cells: {
                    name: {
                        value: 'Vincent Guilmette',
                        className: 'add-custom-class-cell',
                        isHeader: true
                    },
                    id: {
                        value: '111 222 333',
                        className: 'test-cell-custom-class-name'
                    },
                    email: {
                        value: 'vincent.gui@gmail.com'
                    },
                    dateOfBirth: {
                        value: '1976-12-05'
                    },
                    phoneNumber: {
                        value: '819 232-3754'
                    }
                }
            }
        ]
    }),
    props: {
        rowsStyle: {
            default: select(
                'Prop rows-style',
                Enums.toValueArray(MTableRowsStyle),
                MTableRowsStyle.Borders
            )
        }
    },
    template: `<${RESPONSIVE_TABLE_NAME}
        id="Cells"
        :columns="columns"
        :rows="rows"
        :rows-style="rowsStyle"
    />`
});
