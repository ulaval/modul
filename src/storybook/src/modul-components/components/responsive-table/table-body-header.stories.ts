import { boolean, object, select, text } from '@storybook/addon-knobs';
import { RESPONSIVE_TABLE_NAME, TABLE_GROUP_HEADER_NAME } from '@ulaval/modul-components/dist/components/component-names';
import { MTableColumn, MTableEmptyArea, MTableGroup, MTableGroupHeaderStyle } from '@ulaval/modul-components/dist/components/responsive-table/responsive-table-commons';
import { Enums } from '@ulaval/modul-components/dist/utils/enums/enums';
import { modulComponentsHierarchyRootSeparator } from '../../../utils';
import { DEFAULT_TABLE_COLUMNS, DEFAULT_TABLE_GROUP_1 } from './responsive-table-data';

export default {
    title: `${modulComponentsHierarchyRootSeparator}${RESPONSIVE_TABLE_NAME}/${TABLE_GROUP_HEADER_NAME}`,
    parameters: { fileName: __filename }
};

export const defaultStory = () => ({
    data: () => ({
        columns: DEFAULT_TABLE_COLUMNS,
        rowsGroup: DEFAULT_TABLE_GROUP_1
    }),
    template: `<table>
        <${TABLE_GROUP_HEADER_NAME}
            :columns="columns"
            :rowsGroup="rowsGroup"
        />
   </table>`
});

defaultStory.story = {
    name: 'default'
};

export const Sandbox = () => ({
    props: {
        firstColumnFixed: {
            default: boolean('Prop first-column-fixed', false)
        },
        tableComponentWidth: {
            default: text('Prop table-component-width', '100%')
        },
        groupHeaderStyle: {
            default: select(
                'Prop group-header-style',
                Enums.toValueArray(MTableGroupHeaderStyle),
                MTableGroupHeaderStyle.Light
            )
        },
        columns: {
            default: object<MTableColumn[]>('Prop columns', DEFAULT_TABLE_COLUMNS)
        },
        rowsGroup: {
            default: object<MTableGroup>('Prop rows-group', DEFAULT_TABLE_GROUP_1)
        },
        hasDataRowsGroup: {
            default: boolean('has data rows group', true)
        },
        slotBodyHeaderTitle: {
            default: boolean('Slot body-header-title', false)
        },
        slotBodyHeader: {
            default: boolean('Slot body-header', false)
        },
        defaultEmptyArea: {
            default: object<MTableEmptyArea>('Prop default-empty-area', {
                headerText: 'Default empty area: headerText',
                text: 'Default empty area: text',
                iconName: 'm-svg__clock'
            })
        }
    },
    computed: {
        rowsGroupIntern(): MTableGroup[] {
            const _this: any = this;
            return _this.hasDataRowsGroup ? _this.rowsGroup : [];
        }
    },
    template: `<table style="width: 100%">
        <${TABLE_GROUP_HEADER_NAME}
            :columns="columns"
            :rows-group="rowsGroupIntern"
            :first-column-fixed="firstColumnFixed"
            :group-header-style="groupHeaderStyle"
            :default-empty-area="defaultEmptyArea"
            :table-component-width="tableComponentWidth"
        >
            <template
                v-if="slotBodyHeaderTitle && rowsGroup && rowsGroup.header && rowsGroup.header.title"
                slot="body-header-title"
                slot-scope="{ rowsGroup }">
                {{ rowsGroup.header.title}} (Slot body-header-title)
            </template>
            <template
                v-if="slotBodyHeader"
                v-for="(column, columnIndex) in columns"
                :slot="'body-header.' + column.id"
                slot-scope="{ rowsGroup }"
            >
                {{ rowsGroup.header.cells[column.id].value}} (Slot body-header.{{column.id}})
            </template>
        </${TABLE_GROUP_HEADER_NAME}>
    </table>`
});
