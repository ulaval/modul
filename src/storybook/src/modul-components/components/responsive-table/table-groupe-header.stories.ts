import { actions } from '@storybook/addon-actions';
import { boolean, object, select, text } from '@storybook/addon-knobs';
import { RESPONSIVE_TABLE_NAME, TABLE_GROUP_HEADER_NAME } from '@ulaval/modul-components/dist/components/component-names';
import { MTableColumn, MTableEmptyArea, MTableGroupHeaderStyle, MTableRowsGroup } from '@ulaval/modul-components/dist/components/responsive-table/responsive-table-commons';
import { Enums } from '@ulaval/modul-components/dist/utils/enums/enums';
import { ModulIconName } from '@ulaval/modul-components/dist/utils/modul-icons/modul-icons';
import { modulComponentsHierarchyRootSeparator } from '../../../utils';
import { DEFAULT_TABLE_COLUMNS, DEFAULT_TABLE_ROWS_GROUP_1, getScopeSlotTemplate } from './responsive-table-data';

export default {
    title: `${modulComponentsHierarchyRootSeparator}${RESPONSIVE_TABLE_NAME}/${TABLE_GROUP_HEADER_NAME}`,
    parameters: { fileName: __filename }
};

export const Sandbox = () => ({
    props: {
        firstColumnFixed: {
            default: boolean('Prop first-column-fixed', false)
        },
        tableComponentWidth: {
            default: text('Prop table-component-width', '100%')
        },
        groupHeaderClassName: {
            default: text('Prop group-header-class-name', '')
        },
        groupHeaderStyle: {
            default: select(
                'Prop group-header-style',
                Enums.toValueArray(MTableGroupHeaderStyle),
                MTableGroupHeaderStyle.Light
            )
        },

        hasDataRowsGroup: {
            default: boolean('has data rows group', true)
        },
        slotGroupHeaderTitle: {
            default: boolean('Slot group-header-title', false)
        },
        slotGroupHeaderCell: {
            default: boolean('Slot group-header-cell', false)
        },
        slotGroupHeaderCellWithColumn: {
            default: boolean('Slot group-header-cell.<column.name>', false)
        },
        defaultEmptyArea: {
            default: object<MTableEmptyArea>('Prop default-empty-area', {
                title: 'Default empty area: text',
                svgName: ModulIconName.HelpCircle
            })
        },
        columns: {
            default: object<MTableColumn[]>('Prop columns', DEFAULT_TABLE_COLUMNS)
        },
        rowsGroup: {
            default: object<MTableRowsGroup>('Prop rows-group', DEFAULT_TABLE_ROWS_GROUP_1)
        }
    },
    computed: {
        rowsGroupIntern(): MTableRowsGroup[] {
            const _this: any = this;
            return _this.hasDataRowsGroup ? _this.rowsGroup : [];
        }
    },
    methods: actions(
        'emitOpenAccordion',
        'emitCloseAccordion'
    ),
    template: `<table style="width: 100%">
        <${TABLE_GROUP_HEADER_NAME}
            :columns="columns"
            :rows-group="rowsGroup"
            :first-column-fixed="firstColumnFixed"
            :group-header-style="groupHeaderStyle"
            :group-header-class-name="groupHeaderClassName"
            :default-empty-area="defaultEmptyArea"
            :table-component-width="tableComponentWidth"
            @open-accordion="emitOpenAccordion"
            @close-accordion="emitCloseAccordion"
        >
            <template
                v-if="slotGroupHeaderTitle"
                #group-header-title="{ rowsGroup, title, column }">
                ${ getScopeSlotTemplate('{{ title}} (Slot group-header-title)')}
            </template>
            <template
                v-if="slotGroupHeaderCell"
                #group-header-cell="{ rowsGroup, header, cell, column }"
            >
                ${ getScopeSlotTemplate('{{ cell.value }} (Slot group-header-cell)')}
            </template>
            <template
                v-if="slotGroupHeaderCellWithColumn"
                v-for="(column, columnIndex) in columns"
                :slot="'group-header-cell.' + column.name"
                slot-scope="{ rowsGroup, header, cell, column }"
            >
                ${ getScopeSlotTemplate('{{ cell.value }} (Slot group-header-cell.{{column.name}})')}
            </template>
        </${TABLE_GROUP_HEADER_NAME}>
    </table>`
});
