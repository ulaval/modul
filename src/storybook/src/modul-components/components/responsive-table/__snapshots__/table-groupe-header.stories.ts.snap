// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyshots modul-components|m-responsive-table/m-table-group-header Sandbox 1`] = `
<table
  style="width: 100%;"
>
  <tr
    class="m-table-group-header m--has-accordion m--is-accordion-icon-position-left m--is-group-header-style-light"
    tabindex="0"
  >
    <th
      class="m--is-text-align-left"
    >
      <div
        class="m-table-group-header__cell"
      >
        <span
          class="m-plus m-table-group-header__accordion-icon m--is-skin-current-color m--is-open m--is-large add-class-to-icon"
        />
         
        <div
          class="m-table-group-header__content-cell"
        >
          
                            English Class
                         
                                ANL-3010
                            
        </div>
      </div>
    </th>
    <th
      class="m--is-text-align-right"
      colspan="4"
    >
      <div
        class="m-table-group-header__cell"
      >
        <!---->
         
        <div
          class="m-table-group-header__content-cell"
        >
          <!---->
           
                                Advanced English I
                            
        </div>
      </div>
    </th>
    <th
      class="m--is-text-align-left"
    >
      <!---->
    </th>
    <th
      class="m--is-text-align-left"
    >
      <!---->
    </th>
    <th
      class="m--is-text-align-left"
    >
      <!---->
    </th>
  </tr>
</table>
`;
